### PR TITLE
zipdepth: vendor ZipDepth into monopriors (ZipDepthPredictor) + training package

### PR DESCRIPTION
**Pull request by pablo-rerun**
Stack 1/3 — next: optional confidence + SegmentationImage mask; then python-conventions typing pass over the vendored code.

## Part A — `ZipDepthPredictor` in monopriors
- `monopriors/third_party/zipdepth/` — `architecture.py`/`model_utils.py`/`colormap.py` byte-identical to [pablovela5620/ZipDepth@5a80354](https://github.com/pablovela5620/ZipDepth) (== upstream [fabiotosi92/ZipDepth](https://github.com/fabiotosi92/ZipDepth), ECCV 2026, MIT); `predictor.py` only has its `zipdepth.*` imports made relative. **Single copy of the network in the monorepo.**
- `ZipDepthPredictor`: HF weights `pablovela5620/zipdepth` pinned to `0efcd73`; optional local `checkpoint` (trained `final_model.pth` loads through `strip_state_dict_prefixes`); single-owner device move; float32 intrinsics. Registered in `RELATIVE_PREDICTORS`. monopriors `0.3.2 → 0.3.3`.

## Part B — `packages/zipdepth` (training)
- Upstream `data/loss/training/evaluation` unmodified; fork `scripts/` → `zipdepth/upstream_cli/` (import rewrites only) with thin `tools/` shims. No `zipdepth/model` or `model_utils` here (test enforces it).
- Typed tyro `apis/`: `smoke_data`, `infer_rerun` (`RerunTyroConfig` + `log_relative_pred`), `train_smoke` (data → train → resume → torchrun → strict checkpoint round-trip, atomic `latest` rename).
- pixi envs `zipdepth` / `zipdepth-dev`; tasks `zipdepth-infer-rerun`, `zipdepth-train-smoke`. pyrefly: `apis` typed, upstream tree excluded as a vendor boundary (no baseline file).

## Validation (RTX 5090)
- `monoprior-relative-depth --predictor-name ZipDepthPredictor` and `zipdepth-infer-rerun` render RGB / depth / cloud in a headless viewer — report with screenshots: https://pablos-4800gt.ilish-ruler.ts.net:8768/zipdepth/zipdepth-viewer-validation.html. A pixel-level test (`tests/test_infer_rerun_pixels.py`) now guards this.
- `im0.jpg` disparity range `[0.000, 0.147]` == fork reference.
- `zipdepth-train-smoke`: 33 samples, 8 batches/epoch, loss 2.1 → 1.1; resume restores scheduler at step 8; torchrun exit 0; both `final_model.pth` load strict and predict through `ZipDepthPredictor`. Run under the beartype dev env.
- lint / typecheck / deadcode / tests green for `monoprior-dev` and `zipdepth-dev`.
- Reviewed with `/simplify-code` (4 angles) + thermo-nuclear review; all blockers addressed (tools are thin shims, no re-export shims, no baseline, canonical helpers reused, atomic smoke orchestration).

## Phase-2 debt
- Convert `upstream_cli` argparse scripts to tyro `apis/`; strip unused training code.
- Upstream bugs (report-only): `balanced_sampling` config silently ignored (argparse default `False`), dead `logging.log_interval_steps`, no CUDA-availability check in `DepthInference`.
- Not applied from review: RGB→BGR→RGB double swap (would modify vendored `DepthInference`); predictor registry refactor in `relative_depth/__init__.py` (pre-existing pattern shared by all predictors).

### CODE REVIEW

File: packages/monoprior/monopriors/third_party/zipdepth/colormap.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+"""Visualization utilities"""
```
Comment (pablovela5620) [inline packages/monoprior/monopriors/third_party/zipdepth/colormap.py]: why do we need this lut/depth to colomap stuff? dont we already do this fine in rerun?

File: packages/monoprior/monopriors/third_party/zipdepth/predictor.py [hunk 2]
```diff
@@ -0,0 +1,668 @@
+"""
```
Comment (pablovela5620) [inline packages/monoprior/monopriors/third_party/zipdepth/predictor.py]: same here, we're doing inference in our own way. Any readon to include this?

File: packages/monoprior/monopriors/third_party/zipdepth/colormap.py (hunk 1 above)
Comment (pablo-rerun) [reply packages/monoprior/monopriors/third_party/zipdepth/colormap.py]: Agreed — Rerun colormaps for us. Removed `colormap.py` (it was only used by the upstream `DepthInference` save/video paths, which are gone too).

File: packages/monoprior/monopriors/third_party/zipdepth/predictor.py (hunk 2 above)
Comment (pablo-rerun) [reply packages/monoprior/monopriors/third_party/zipdepth/predictor.py]: No good reason. Dropped the vendored `DepthInference`; `ZipDepthPredictor` now owns the ~15 lines of preprocessing (shorter side → `input_size` rounded to /32, `/255`, forward, bilinear resize back — the model normalizes internally) and loads checkpoints strictly via `load_zipdepth()`. `third_party/zipdepth/` is now just `architecture.py` + `model_utils.py`, byte-identical to the fork. Output is unchanged (`im0.jpg` disparity `[0.000, 0.147]`); smoke pipeline re-run from scratch passes.
